### ISSUE TITLE
removeFile does not work for file input with initial value

### ISSUE THREAD

**Issue by hamblin** [wearebraid/vue-formulate#265]
**Describe the bug**
If a file input has an initial value set, then you click on the X to remove it (calling removeFile on the FileUpload class), the v-model data from that input is not cleared.  

If you upload a new file in its place, the new file data does take its place (good) but there are use cases where a user wants to load a form with existing data and remove a file previously uploaded, not uploading anything new in its place.  At a minimum, I would expect the model data to be cleared when user clicks to remove file.  (of course what happens on the server side for this is out of scope here)

Same issue existed on < 2.4.4 when removing files that were uploaded in the same session, and I saw that was fixed in 2.4.4 with some changes to the FileUpload class. (great!)  I think the same solution used there could be used for this situation as well.  

I reproduced the issues in this CodePen:  https://codepen.io/hamblin/pen/XWdwRye

This may be related to issue #245 -- that pertained to files newly uploaded, I believe.  This relates to files passed in as initial values.

**To Reproduce**
1. Go to this CodePen:  https://codepen.io/hamblin/pen/XWdwRye  which has an initial value set for the file input (:value=[{/path/to/some/file.jpg}].  The current value of the file input is shown on the right side (updated via @input event on the file input)
2. Click on the X to remove that file. 
3. Observe that the file input value is not cleared... same url: '/file/path...' is still there.  <-- this is the issue
4. Upload a new file instead... file input value gets updated as expected.  (no issue, works as expected)
5. Click on the X again to remove that NEW file -- file input value gets cleared as expected.  (no issue, works as expected)

**Reproduction**
See above, and linked CodePen.

### CodePen:
https://codepen.io/hamblin/pen/XWdwRye

**Expected behavior**
I expected the 'remove file' behavior to work the same for both newly uploaded files, and for files that were passed in as initial values.  It seems the 'removeFile' fix in 2.4.4 only worked for newly uploaded files, not for files passed as initial values.  

Looking at my Vue devtools, I see that when removeFile is clicked for a initial value file, the file DOES get removed from the 'files' array, but **not removed** from the (faux) fileList and results arrays.  See screenshot below.

One idea for a fix... the setId function that was used to set nanoId/uuid on each file in 2.4.4 -- perhaps that could be used also in the rehydrateFileList and addFileList methods of FileUpload class to add the nanoId to files in those arrays as well when loading initial files.  Then existing removeFile method should remove them as written in 2.4.4, since using the nanoId how it seems to look for the files to remove, I think.  

**Screenshots**
Screenshot below shows my Vue devtools for the file input value AFTER clicking remove file on a file passed in as initial value.  Note that the file was removed from 'files' array, but not fileList or results arrays.
![vue-formulate-initial-file-value](https://user-images.githubusercontent.com/17462329/94622250-269a1b00-02b2-11eb-88cc-fe8155ab9737.png)

**Device information:**
 - 2020 MacBookPro
 - OS: 10.15.6
 - Browser: Chrome 85
 - Version:  vue-formulate 2.4.4  (originally found on 2.4.3, but reproduced on 2.4.4)

Thanks to the makers / maintainers of vue-formulate... its really awesome!

**Comment by justin-schroeder** [comment]
@hamblin 👋

Thanks for the great write up, it made it very clear where to track down the issue. A few annoying things at play here:

1. JSDom does not support FileList so we cannot write vue tests for this specific piece of functionality which is super annoying. Everything in Vue Formulate is heavily tested except for these FileList "black holes". Eventually I'd like to add cypress to the mix just to test these file inputs.
2. FileList is just super annoying to work with 😂. You cannot just create instance of it willy nilly (thus the [DataTrasfer](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer) which is technically for drag/drop events). So anywhere we touch the file list we need to branch the logic to check if it's actually an instance of FileList.
3. That branched logic is missing the "remove" functionality if the `fileList` is _not_ a `FileList` instance. Ooops. Check it out here: https://github.com/wearebraid/vue-formulate/blob/master/src/FileUpload.js#L175-L179

So great find, honestly I’m pretty surprised this hasn't come up sooner. Tagging this as a bug. Should be easy to fix :)

**Comment by hamblin** [comment]
Thanks for that great explanation & detail, Justin.  I did see that DataTransfer in the code... then scratched my head a bit and assumed there was a good reason for it that goes above and beyond my newbie mind.  (rightly so!)  

I already made a higher-level workaround in my app, but will be happy to test out any fixes.  (but maybe not even needed)  

And thanks again for making Vue Formulate... I'm honestly kicking myself for not finding it sooner.
